tsl/src/remote/txn.c

```diff
@@ -90,6 +90,7 @@ remote_txn_begin(RemoteTxn *entry, int curlevel)
 	{
 		const char *sql;
 
+		Assert(remote_connection_get_status(entry->conn) == CONN_IDLE);
```

Comment:
If `remote_connection_get_status` has side-effects, that might hide bugs later in the code that surface once the assertion is removed.

Reply:
Why do you think `remote_connection_get_status` has side effects? It is just a simple "getter".

Reply:
It is not a problem in the current code, but it might be the result of a future change, e.g., reading the status code from a receive buffer.

Reply:
I am not sure I follow. And why is reading status from a buffer a side-effect (unless it also modifies the buffer)?